src/cryptography/hazmat/backends/openssl/backend.py

```diff
@@ -686,6 +776,68 @@ def cmac_algorithm_supported(self, algorithm):
     def create_cmac_ctx(self, algorithm):
         return _CMACContext(self, algorithm)
 
+    def create_x509_csr(self, builder, private_key, algorithm):
+        # TODO: check type of private key parameter.
```

Comment:
Not every private_key type is guaranteed to have an `_evp_pkey` right now (and, in fact, only RSA does). There has been past discussion in #1231 about having all our OpenSSL asym objects take evp_pkey, so maybe we should do that so this doesn't need to know anything about type. Alternately this will need to check type and create an evp_pkey for DSA/EC.

Either way, we'll need at least one CSR generation for EC and DSA to confirm this works (which right now it won't).